lib/fswatcher/fswatcher.go

```diff
@@ -108,14 +115,23 @@ func (watcher *FsWatcher) watchFilesystem() {
 	}
 }
 
-func (watcher *FsWatcher) newFsEvent(eventPath string) *FsEvent {
-	if isSubpath(eventPath, watcher.folderPath) {
+func (watcher *FsWatcher) newFsEvent(eventPath string) {
+	if len(watcher.fsEvents) == maxFiles {
+		watcher.debugf("Tracking too many events; dropping: %s\n", eventPath)
```

Comment:
We do [exactly that in actOnTimer](https://github.com/syncthing/syncthing/pull/2807/files/04bb0ed836d46a1b8ba19b764a1794e03f33e3f4#diff-b32903d990e7fbffc458478f1a0db30fR225).